HPX 1.2.0 does not build on i686, but release candidate did

## Expected Behavior

Should compile as on x86, but does not on i686. Interesting fact is that the release candidate builds on i686, but not HPX 1.2.0.

## Actual Behavior

```
make[2]: Entering directory '/builddir/build/BUILD/hpx_1.2.0/serial'
[ 96%] Building CXX object examples/resource_partitioner/CMakeFiles/async_customization_exe.dir/async_customization.cpp.o
cd /builddir/build/BUILD/hpx_1.2.0/serial/examples/resource_partitioner && /usr/bin/c++  -DBOOST_DISABLE_ASSERTS -DHPX_APPLICATION_EXPORTS -DHPX_APPLICATION_NAME=async_customization_exe -DHPX_APPLICATION_STRING=\"async_customization_exe\" -DHPX_DISABLE_ASSERTS -DHPX_PREFIX=\"/usr\" -D_GNU_SOURCE -I/builddir/build/BUILD/hpx_1.2.0 -I/builddir/build/BUILD/hpx_1.2.0/serial -I/builddir/build/BUILD/hpx_1.2.0/tests -I/builddir/build/BUILD/hpx_1.2.0/examples  -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m32 -march=i686 -mtune=generic -msse2 -mfpmath=sse -mstackrealign -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -DNDEBUG   -std=c++17 -pthread -Wall -Wextra -Wno-unused-local-typedefs -Wno-strict-aliasing -Wno-sign-promo -Wno-attributes -Wno-cast-align -Wno-unused-parameter -Wformat=2 -Wno-format-nonliteral -Winit-self -Wdouble-promotion -Wcast-qual -Wcast-align -Werror=trampolines -Werror=parentheses -Werror=reorder -Werror=return-type -Werror=sequence-point -Werror=uninitialized -Werror=format -Werror=missing-braces -Werror=sign-compare -fdiagnostics-show-option -Werror=vla -Wno-unused-but-set-parameter -Wno-unused-but-set-variable -Wno-maybe-uninitialized -Wno-sync-nand -Wno-delete-non-virtual-dtor -fvisibility=hidden -o CMakeFiles/async_customization_exe.dir/async_customization.cpp.o -c /builddir/build/BUILD/hpx_1.2.0/examples/resource_partitioner/async_customization.cpp
In file included from /builddir/build/BUILD/hpx_1.2.0/examples/resource_partitioner/async_customization.cpp:9:
/builddir/build/BUILD/hpx_1.2.0/hpx/runtime/threads/executors/guided_pool_executor.hpp: In instantiation of 'auto hpx::threads::executors::pre_execution_then_domain_schedule<Executor, NumaFunction>::operator()(F&&, Future&&, Ts&& ...) const [with F = const test(Executor&) [with Executor = hpx::threads::executors::guided_pool_executor_shim<hpx::threads::executors::pool_numa_hint<dummy_tag> >]::<lambda(hpx::lcos::future<hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> > >&&)>; Future = hpx::lcos::future<hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> > >; Ts = {}; Executor = hpx::threads::executors::pool_executor; NumaFunction = hpx::threads::executors::pool_numa_hint<dummy_tag>]':
/builddir/build/BUILD/hpx_1.2.0/hpx/runtime/threads/executors/guided_pool_executor.hpp:385:36:   required from 'hpx::lcos::future<typename hpx::util::detail::invoke_deferred_result<F, OuterFuture<hpx::util::tuple<InnerFutures ...> >, Ts ...>::type> hpx::threads::executors::guided_pool_executor<hpx::threads::executors::pool_numa_hint<Tag> >::then_execute(F&&, OuterFuture<hpx::util::tuple<InnerFutures ...> >&&, Ts&& ...) [with F = test(Executor&) [with Executor = hpx::threads::executors::guided_pool_executor_shim<hpx::threads::executors::pool_numa_hint<dummy_tag> >]::<lambda(hpx::lcos::future<hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> > >&&)>; OuterFuture = hpx::lcos::future; InnerFutures = {hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float>}; Ts = {}; <template-parameter-2-5> = void; <template-parameter-2-6> = void; Tag = dummy_tag; typename hpx::util::detail::invoke_deferred_result<F, OuterFuture<hpx::util::tuple<InnerFutures ...> >, Ts ...>::type = std::complex<double>]'
/builddir/build/BUILD/hpx_1.2.0/hpx/runtime/threads/executors/guided_pool_executor.hpp:555:40:   required from 'hpx::lcos::future<typename hpx::util::detail::invoke_deferred_result<F, Future, Ts ...>::type> hpx::threads::executors::guided_pool_executor_shim<H>::then_execute(F&&, Future&&, Ts&& ...) [with F = test(Executor&) [with Executor = hpx::threads::executors::guided_pool_executor_shim<hpx::threads::executors::pool_numa_hint<dummy_tag> >]::<lambda(hpx::lcos::future<hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> > >&&)>; Future = hpx::lcos::future<hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> > >; Ts = {}; <template-parameter-2-4> = void; H = hpx::threads::executors::pool_numa_hint<dummy_tag>; typename hpx::util::detail::invoke_deferred_result<F, Future, Ts ...>::type = std::complex<double>]'
/builddir/build/BUILD/hpx_1.2.0/hpx/parallel/executors/execution.hpp:446:44:   required from 'static decltype (exec.then_execute(forward<F>(f), forward<Future>(predecessor), (forward<Ts>)(hpx::parallel::execution::detail::then_execute_fn_helper<Executor, typename std::enable_if<hpx::traits::is_two_way_executor<T, void>::value, void>::type>::call_impl::ts)...)) hpx::parallel::execution::detail::then_execute_fn_helper<Executor, typename std::enable_if<hpx::traits::is_two_way_executor<T>::value>::type>::call_impl(int, TwoWayExecutor&&, F&&, Future&&, Ts&& ...) [with TwoWayExecutor = hpx::threads::executors::guided_pool_executor_shim<hpx::threads::executors::pool_numa_hint<dummy_tag> >&; F = test(Executor&) [with Executor = hpx::threads::executors::guided_pool_executor_shim<hpx::threads::executors::pool_numa_hint<dummy_tag> >]::<lambda(hpx::lcos::future<hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> > >&&)>; Future = hpx::lcos::future<hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> > >; Ts = {}; Executor = hpx::threads::executors::guided_pool_executor_shim<hpx::threads::executors::pool_numa_hint<dummy_tag> >; decltype (exec.then_execute(forward<F>(f), forward<Future>(predecessor), (forward<Ts>)(hpx::parallel::execution::detail::then_execute_fn_helper<Executor, typename std::enable_if<hpx::traits::is_two_way_executor<T, void>::value, void>::type>::call_impl::ts)...)) = hpx::lcos::future<std::complex<double> >]'
/builddir/build/BUILD/hpx_1.2.0/hpx/parallel/executors/execution.hpp:459:33:   required from 'static decltype (hpx::parallel::execution::detail::then_execute_fn_helper<Executor, typename std::enable_if<hpx::traits::is_two_way_executor<T>::value>::type>::call_impl(0, forward<TwoWayExecutor>(exec), forward<F>(f), forward<Future>(predecessor), (forward<Ts>)(hpx::parallel::execution::detail::then_execute_fn_helper<Executor, typename std::enable_if<hpx::traits::is_two_way_executor<T, void>::value, void>::type>::call::ts)...)) hpx::parallel::execution::detail::then_execute_fn_helper<Executor, typename std::enable_if<hpx::traits::is_two_way_executor<T>::value>::type>::call(TwoWayExecutor&&, F&&, Future&&, Ts&& ...) [with TwoWayExecutor = hpx::threads::executors::guided_pool_executor_shim<hpx::threads::executors::pool_numa_hint<dummy_tag> >&; F = test(Executor&) [with Executor = hpx::threads::executors::guided_pool_executor_shim<hpx::threads::executors::pool_numa_hint<dummy_tag> >]::<lambda(hpx::lcos::future<hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> > >&&)>; Future = hpx::lcos::future<hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> > >; Ts = {}; Executor = hpx::threads::executors::guided_pool_executor_shim<hpx::threads::executors::pool_numa_hint<dummy_tag> >; decltype (hpx::parallel::execution::detail::then_execute_fn_helper<Executor, typename std::enable_if<hpx::traits::is_two_way_executor<T>::value>::type>::call_impl(0, forward<TwoWayExecutor>(exec), forward<F>(f), forward<Future>(predecessor), (forward<Ts>)(hpx::parallel::execution::detail::then_execute_fn_helper<Executor, typename std::enable_if<hpx::traits::is_two_way_executor<T, void>::value, void>::type>::call::ts)...)) = hpx::lcos::future<std::complex<double> >]'
/builddir/build/BUILD/hpx_1.2.0/hpx/parallel/executors/execution_fwd.hpp:312:24:   [ skipping 2 instantiation contexts, use -ftemplate-backtrace-limit=0 to disable ]
/builddir/build/BUILD/hpx_1.2.0/hpx/lcos/local/packaged_continuation.hpp:680:49:   required from 'typename hpx::traits::future_then_executor_result<Executor, typename std::decay<_Arg>::type, F>::type hpx::lcos::detail::then_execute_helper(Executor&&, F&&, Future&&) [with Executor = hpx::threads::executors::guided_pool_executor_shim<hpx::threads::executors::pool_numa_hint<dummy_tag> >&; Future = hpx::lcos::future<hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> > >; F = test(Executor&) [with Executor = hpx::threads::executors::guided_pool_executor_shim<hpx::threads::executors::pool_numa_hint<dummy_tag> >]::<lambda(hpx::lcos::future<hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> > >&&)>; typename hpx::traits::future_then_executor_result<Executor, typename std::decay<_Arg>::type, F>::type = hpx::lcos::future<std::complex<double> >]'
/builddir/build/BUILD/hpx_1.2.0/hpx/lcos/future.hpp:477:47:   required from 'static typename hpx::traits::future_then_executor_result<Executor_, Future, F>::type hpx::lcos::detail::future_then_dispatch<Future, Executor, typename std::enable_if<((hpx::traits::is_one_way_executor<Executor>::value || hpx::traits::is_two_way_executor<Executor>::value) || hpx::traits::is_threads_executor<Executor>::value)>::type>::call(Future&&, Executor_&&, F&&) [with Executor_ = hpx::threads::executors::guided_pool_executor_shim<hpx::threads::executors::pool_numa_hint<dummy_tag> >&; F = test(Executor&) [with Executor = hpx::threads::executors::guided_pool_executor_shim<hpx::threads::executors::pool_numa_hint<dummy_tag> >]::<lambda(hpx::lcos::future<hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> > >&&)>; Future = hpx::lcos::future<hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> > >; Executor = hpx::threads::executors::guided_pool_executor_shim<hpx::threads::executors::pool_numa_hint<dummy_tag> >; typename hpx::traits::future_then_executor_result<Executor_, Future, F>::type = hpx::lcos::future<std::complex<double> >]'
/builddir/build/BUILD/hpx_1.2.0/hpx/lcos/future.hpp:725:77:   required from 'static decltype (hpx::lcos::detail::future_base<Derived, R>::future_then_dispatch<typename std::decay<T0>::type>::call(std::move(fut), forward<T0>(t0), forward<F>(f))) hpx::lcos::detail::future_base<Derived, R>::then(Derived&&, T0&&, F&&, hpx::error_code&) [with F = test(Executor&) [with Executor = hpx::threads::executors::guided_pool_executor_shim<hpx::threads::executors::pool_numa_hint<dummy_tag> >]::<lambda(hpx::lcos::future<hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> > >&&)>; T0 = hpx::threads::executors::guided_pool_executor_shim<hpx::threads::executors::pool_numa_hint<dummy_tag> >&; Derived = hpx::lcos::future<hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> > >; R = hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> >; decltype (hpx::lcos::detail::future_base<Derived, R>::future_then_dispatch<typename std::decay<T0>::type>::call(std::move(fut), forward<T0>(t0), forward<F>(f))) = hpx::lcos::future<std::complex<double> >; typename std::decay<T0>::type = hpx::threads::executors::guided_pool_executor_shim<hpx::threads::executors::pool_numa_hint<dummy_tag> >]'
/builddir/build/BUILD/hpx_1.2.0/hpx/lcos/future.hpp:1069:35:   required from 'typename hpx::util::lazy_enable_if<((hpx::traits::is_one_way_executor<typename std::decay<_U1>::type>::value || hpx::traits::is_two_way_executor<typename std::decay<_U1>::type>::value) || hpx::traits::is_threads_executor<typename std::decay<_U1>::type>::value), hpx::traits::future_then_executor_result<Executor, hpx::lcos::future<R>, F> >::type hpx::lcos::future<R>::then(Executor&&, F&&, hpx::error_code&) [with Executor = hpx::threads::executors::guided_pool_executor_shim<hpx::threads::executors::pool_numa_hint<dummy_tag> >&; F = test(Executor&) [with Executor = hpx::threads::executors::guided_pool_executor_shim<hpx::threads::executors::pool_numa_hint<dummy_tag> >]::<lambda(hpx::lcos::future<hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> > >&&)>; R = hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> >; typename hpx::util::lazy_enable_if<((hpx::traits::is_one_way_executor<typename std::decay<_U1>::type>::value || hpx::traits::is_two_way_executor<typename std::decay<_U1>::type>::value) || hpx::traits::is_threads_executor<typename std::decay<_U1>::type>::value), hpx::traits::future_then_executor_result<Executor, hpx::lcos::future<R>, F> >::type = hpx::lcos::future<std::complex<double> >]'
/builddir/build/BUILD/hpx_1.2.0/examples/resource_partitioner/async_customization.cpp:388:10:   required from 'int test(Executor&) [with Executor = hpx::threads::executors::guided_pool_executor_shim<hpx::threads::executors::pool_numa_hint<dummy_tag> >]'
/builddir/build/BUILD/hpx_1.2.0/examples/resource_partitioner/async_customization.cpp:480:21:   required from here
/builddir/build/BUILD/hpx_1.2.0/hpx/runtime/threads/executors/guided_pool_executor.hpp:161:40: error: no match for call to '(hpx::threads::executors::pool_numa_hint<dummy_tag>) (const hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> >&)'
             int domain = numa_function_(predecessor_value, ts...);
                          ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~
/builddir/build/BUILD/hpx_1.2.0/examples/resource_partitioner/async_customization.cpp:448:11: note: candidate: 'int hpx::threads::executors::pool_numa_hint<dummy_tag>::operator()(int, double, const char*) const'
       int operator()(const int, const double, const char *) const {
           ^~~~~~~~
/builddir/build/BUILD/hpx_1.2.0/examples/resource_partitioner/async_customization.cpp:448:11: note:   candidate expects 3 arguments, 1 provided
/builddir/build/BUILD/hpx_1.2.0/examples/resource_partitioner/async_customization.cpp:452:11: note: candidate: 'int hpx::threads::executors::pool_numa_hint<dummy_tag>::operator()(int) const'
       int operator()(const int ) const {
           ^~~~~~~~
/builddir/build/BUILD/hpx_1.2.0/examples/resource_partitioner/async_customization.cpp:452:11: note:   no known conversion for argument 1 from 'const hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> >' to 'int'
/builddir/build/BUILD/hpx_1.2.0/examples/resource_partitioner/async_customization.cpp:456:11: note: candidate: 'int hpx::threads::executors::pool_numa_hint<dummy_tag>::operator()(uint16_t, double) const'
       int operator()(const uint16_t, const double) const {
           ^~~~~~~~
/builddir/build/BUILD/hpx_1.2.0/examples/resource_partitioner/async_customization.cpp:456:11: note:   candidate expects 2 arguments, 1 provided
/builddir/build/BUILD/hpx_1.2.0/examples/resource_partitioner/async_customization.cpp:460:11: note: candidate: 'int hpx::threads::executors::pool_numa_hint<dummy_tag>::operator()(const hpx::util::tuple<hpx::lcos::future<int>, hpx::lcos::future<double> >&) const'
       int operator()(const util::tuple<future<int>, future<double>> &) const {
           ^~~~~~~~
/builddir/build/BUILD/hpx_1.2.0/examples/resource_partitioner/async_customization.cpp:460:11: note:   no known conversion for argument 1 from 'const hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> >' to 'const hpx::util::tuple<hpx::lcos::future<int>, hpx::lcos::future<double> >&'
/builddir/build/BUILD/hpx_1.2.0/examples/resource_partitioner/async_customization.cpp:463:11: note: candidate: 'int hpx::threads::executors::pool_numa_hint<dummy_tag>::operator()(const hpx::util::tuple<hpx::lcos::future<long unsigned int>, hpx::lcos::shared_future<float> >&) const'
       int operator()(const util::tuple<future<long unsigned int>,
           ^~~~~~~~
/builddir/build/BUILD/hpx_1.2.0/examples/resource_partitioner/async_customization.cpp:463:11: note:   no known conversion for argument 1 from 'const hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> >' to 'const hpx::util::tuple<hpx::lcos::future<long unsigned int>, hpx::lcos::shared_future<float> >&'
In file included from /builddir/build/BUILD/hpx_1.2.0/hpx/lcos/local/packaged_continuation.hpp:22,
                 from /builddir/build/BUILD/hpx_1.2.0/hpx/lcos/future.hpp:1798,
                 from /builddir/build/BUILD/hpx_1.2.0/hpx/parallel/executors/default_executor.hpp:12,
                 from /builddir/build/BUILD/hpx_1.2.0/hpx/parallel/executors.hpp:11,
                 from /builddir/build/BUILD/hpx_1.2.0/hpx/include/parallel_executors.hpp:9,
                 from /builddir/build/BUILD/hpx_1.2.0/examples/resource_partitioner/async_customization.cpp:6:
/builddir/build/BUILD/hpx_1.2.0/hpx/util/deferred_call.hpp:160:5: error: 'hpx::util::detail::deferred<typename std::decay<_Tp>::type, typename std::decay<_Args>::type ...> hpx::util::deferred_call(F&&, Ts&& ...) [with F = const test(Executor&) [with Executor = hpx::threads::executors::guided_pool_executor_shim<hpx::threads::executors::pool_numa_hint<dummy_tag> >]::<lambda(hpx::lcos::future<hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> > >&&)>; Ts = {hpx::lcos::future<hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> > >}; typename std::decay<_Tp>::type = test(Executor&) [with Executor = hpx::threads::executors::guided_pool_executor_shim<hpx::threads::executors::pool_numa_hint<dummy_tag> >]::<lambda(hpx::lcos::future<hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> > >&&)>]', declared using local type 'const test(Executor&) [with Executor = hpx::threads::executors::guided_pool_executor_shim<hpx::threads::executors::pool_numa_hint<dummy_tag> >]::<lambda(hpx::lcos::future<hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> > >&&)>', is used but never defined [-fpermissive]
     deferred_call(F&& f, Ts&&... vs)
     ^~~~~~~~~~~~~
In file included from /builddir/build/BUILD/hpx_1.2.0/hpx/parallel/executors/execution.hpp:16,
                 from /builddir/build/BUILD/hpx_1.2.0/hpx/lcos/local/packaged_continuation.hpp:25,
                 from /builddir/build/BUILD/hpx_1.2.0/hpx/lcos/future.hpp:1798,
                 from /builddir/build/BUILD/hpx_1.2.0/hpx/parallel/executors/default_executor.hpp:12,
                 from /builddir/build/BUILD/hpx_1.2.0/hpx/parallel/executors.hpp:11,
                 from /builddir/build/BUILD/hpx_1.2.0/hpx/include/parallel_executors.hpp:9,
                 from /builddir/build/BUILD/hpx_1.2.0/examples/resource_partitioner/async_customization.cpp:6:
/builddir/build/BUILD/hpx_1.2.0/hpx/lcos/dataflow.hpp:531:10: error: 'decltype (hpx::lcos::detail::dataflow_dispatch<typename std::decay<_Tp>::type>::call(hpx::util::internal_allocator<>{}, forward<F>(f), (forward<Ts>)(hpx::dataflow::ts)...)) hpx::dataflow(F&&, Ts&& ...) [with F = const hpx::detail::sync_policy&; Ts = {hpx::threads::executors::guided_pool_executor<hpx::threads::executors::pool_numa_hint<Tag> >::then_execute(F&&, OuterFuture<hpx::util::tuple<InnerFutures ...> >&&, Ts&& ...) [with F = test(Executor&) [with Executor = hpx::threads::executors::guided_pool_executor_shim<hpx::threads::executors::pool_numa_hint<dummy_tag> >]::<lambda(hpx::lcos::future<hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> > >&&)>; OuterFuture = hpx::lcos::future; InnerFutures = {hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float>}; Ts = {}; <template-parameter-2-5> = void; <template-parameter-2-6> = void; Tag = dummy_tag; typename hpx::util::detail::invoke_deferred_result<F, OuterFuture<hpx::util::tuple<InnerFutures ...> >, Ts ...>::type = std::complex<double>]::<lambda(hpx::lcos::future<hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> > >&&)>, hpx::lcos::future<hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> > >}; decltype (hpx::lcos::detail::dataflow_dispatch<typename std::decay<_Tp>::type>::call(hpx::util::internal_allocator<>{}, forward<F>(f), (forward<Ts>)(hpx::dataflow::ts)...)) = hpx::lcos::future<hpx::lcos::future<std::complex<double> > >; typename std::decay<_Tp>::type = hpx::detail::sync_policy]', declared using local type 'hpx::threads::executors::guided_pool_executor<hpx::threads::executors::pool_numa_hint<Tag> >::then_execute(F&&, OuterFuture<hpx::util::tuple<InnerFutures ...> >&&, Ts&& ...) [with F = test(Executor&) [with Executor = hpx::threads::executors::guided_pool_executor_shim<hpx::threads::executors::pool_numa_hint<dummy_tag> >]::<lambda(hpx::lcos::future<hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> > >&&)>; OuterFuture = hpx::lcos::future; InnerFutures = {hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float>}; Ts = {}; <template-parameter-2-5> = void; <template-parameter-2-6> = void; Tag = dummy_tag; typename hpx::util::detail::invoke_deferred_result<F, OuterFuture<hpx::util::tuple<InnerFutures ...> >, Ts ...>::type = std::complex<double>]::<lambda(hpx::lcos::future<hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> > >&&)>', is used but never defined [-fpermissive]
     auto dataflow(F && f, Ts &&... ts)
          ^~~~~~~~
make[2]: *** [examples/resource_partitioner/CMakeFiles/async_customization_exe.dir/build.make:66: examples/resource_partitioner/CMakeFiles/async_customization_exe.dir/async_customization.cpp.o] Error 1
make[2]: Leaving directory '/builddir/build/BUILD/hpx_1.2.0/serial'
make[1]: *** [CMakeFiles/Makefile2:61761: examples/resource_partitioner/CMakeFiles/async_customization_exe.dir/all] Error 2
make[1]: Leaving directory '/builddir/build/BUILD/hpx_1.2.0/serial'
make: *** [Makefile:144: all] Error 2
```
Full build log (https://kojipkgs.fedoraproject.org//work/tasks/6776/30886776/build.log)

## Steps to Reproduce the Problem

... Please be as specific as possible while describing how to reproduce your problem.

  1.
  1.
  1.

## Specifications

... Please describe your environment

  - HPX Version: 1.2.0
  - Platform (compiler, OS):

```
+ /usr/bin/cmake -DCMAKE_C_FLAGS_RELEASE:STRING=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE:STRING=-DNDEBUG -DCMAKE_Fortran_FLAGS_RELEASE:STRING=-DNDEBUG -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON -DCMAKE_INSTALL_PREFIX:PATH=/usr -DINCLUDE_INSTALL_DIR:PATH=/usr/include -DLIB_INSTALL_DIR:PATH=/usr/lib -DSYSCONF_INSTALL_DIR:PATH=/etc -DSHARE_INSTALL_PREFIX:PATH=/usr/share -DBUILD_SHARED_LIBS:BOOL=ON -DLIB=lib ..
-- The CXX compiler identification is GNU 8.2.1
-- The C compiler identification is GNU 8.2.1
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- CMake version:  3.12.2
-- HPX version:  1.2.0
-- No vectorization library configured
-- Native TLS is enabled.
-- Nice threadlevel is disabled.
-- Stack traces are enabled.
-- Compiling with the native toolset
-- Performing Test HPX_WITH_CXX17
-- Performing Test HPX_WITH_CXX17 - Success
-- C++ mode used: C++17
-- Performing Test HPX_WITH_CXX11_ALIAS_TEMPLATES - Success (cmake feature test)
-- Performing Test HPX_WITH_CXX11_AUTO - Success (cmake feature test)
-- Performing Test HPX_WITH_CXX11_CONSTEXPR - Success (cmake feature test)
-- Performing Test HPX_WITH_CXX11_DECLTYPE - Success (cmake feature test)
-- Performing Test HPX_WITH_CXX11_SFINAE_EXPRESSION - Success
-- Performing Test HPX_WITH_CXX11_DEFAULTED_FUNCTIONS - Success (cmake feature test)
-- Performing Test HPX_WITH_CXX11_DELETED_FUNCTIONS - Success (cmake feature test)
-- Performing Test HPX_WITH_CXX11_EXPLICIT_CONVERSION_OPERATORS - Success (cmake feature test)
-- Performing Test HPX_WITH_CXX11_EXPLICIT_VARIADIC_TEMPLATES - Success
-- Performing Test HPX_WITH_CXX11_EXTENDED_FRIEND_DECLARATIONS - Success (cmake feature test)
-- Performing Test HPX_WITH_CXX11_FUNCTION_TEMPLATE_DEFAULT_ARGS - Success (cmake feature test)
-- Performing Test HPX_WITH_CXX11_INLINE_NAMESPACES - Success (cmake feature test)
-- Performing Test HPX_WITH_CXX11_LAMBDAS - Success (cmake feature test)
-- Performing Test HPX_WITH_CXX11_NOEXCEPT - Success (cmake feature test)
-- Performing Test HPX_WITH_CXX11_NULLPTR - Success (cmake feature test)
-- Performing Test HPX_WITH_CXX11_NSDMI - Success (cmake feature test)
-- Performing Test HPX_WITH_CXX11_RANGE_BASED_FOR - Success (cmake feature test)
-- Performing Test HPX_WITH_CXX11_RVALUE_REFERENCES - Success (cmake feature test)
-- Performing Test HPX_WITH_CXX11_SCOPED_ENUMS - Success (cmake feature test)
-- Performing Test HPX_WITH_CXX11_STATIC_ASSERT - Success (cmake feature test)
-- Performing Test HPX_WITH_CXX11_VARIADIC_MACROS - Success (cmake feature test)
-- Performing Test HPX_WITH_CXX11_VARIADIC_TEMPLATES - Success (cmake feature test)
-- Performing Test HPX_WITH_CXX11_OVERRIDE - Success (cmake feature test)
-- Performing Test HPX_WITH_CXX11_ARRAY - Success
-- Performing Test HPX_WITH_CXX11_ATOMIC - Success
-- Performing Test HPX_WITH_CXX11_CHRONO - Success
-- Performing Test HPX_WITH_CXX11_CSTDINT - Success
-- Performing Test HPX_WITH_CXX11_EXCEPTION_PTR - Success
-- Performing Test HPX_WITH_CXX11_STD_INITIALIZER_LIST - Success
-- Performing Test HPX_WITH_CXX11_IS_BIND_EXPRESSION - Success
-- Performing Test HPX_WITH_CXX11_IS_PLACEHOLDER - Success
-- Performing Test HPX_WITH_CXX11_IS_TRIVIALLY_COPYABLE - Success
-- Performing Test HPX_WITH_CXX11_LOCK_GUARD - Success
-- Performing Test HPX_WITH_CXX11_RANDOM - Success
-- Performing Test HPX_WITH_CXX11_RANGE_ACCESS - Success
-- Performing Test HPX_WITH_CXX11_REFERENCE_WRAPPER - Success
-- Performing Test HPX_WITH_CXX11_SHARED_PTR - Success
-- Performing Test HPX_WITH_CXX11_SHUFFLE - Success
-- Performing Test HPX_WITH_CXX11_THREAD - Success
-- Performing Test HPX_WITH_CXX11_TO_STRING - Success
-- Performing Test HPX_WITH_CXX11_UNIQUE_LOCK - Success
-- Performing Test HPX_WITH_CXX11_UNIQUE_PTR - Success
-- Performing Test HPX_WITH_CXX11_UNORDERED_MAP - Success
-- Performing Test HPX_WITH_CXX11_UNORDERED_SET - Success
-- Performing Test HPX_WITH_CXX11_THREAD_LOCAL - Success (cmake feature test)
-- Performing Test HPX_WITH_CXX11_NORETURN_ATTRIBUTE - Success
-- Performing Test HPX_WITH_CXX14_CONSTEXPR - Success
-- Performing Test HPX_WITH_CXX14_LAMBDAS - Success (cmake feature test)
-- Performing Test HPX_WITH_CXX14_INTEGER_SEQUENCE - Success
-- Performing Test HPX_WITH_CXX14_IS_FINAL - Success
-- Performing Test HPX_WITH_CXX14_IS_NULL_POINTER - Success
-- Performing Test HPX_WITH_CXX14_RESULT_OF_SFINAE - Success
-- Performing Test HPX_WITH_CXX14_VARIABLE_TEMPLATES - Success (cmake feature test)
-- Performing Test HPX_WITH_CXX14_DEPRECATED_ATTRIBUTE - Success
-- Performing Test HPX_WITH_CXX14_RETURN_TYPE_DEDUCTION - Success (cmake feature test)
-- Performing Test HPX_WITH_CXX17_FOLD_EXPRESSIONS - Success
-- Performing Test HPX_WITH_CXX17_FALLTHROUGH_ATTRIBUTE - Success
-- Performing Test HPX_WITH_MM_PREFETCH - Success
-- Performing Test HPX_WITH_STABLE_INPLACE_MERGE - Success
-- Performing Test HPX_WITH_UNISTD_H - Success
-- Performing Test HPX_WITH_CXX_FLAG_PTHREAD
-- Performing Test HPX_WITH_CXX_FLAG_PTHREAD - Success
-- Performing Test HPX_WITH_CXX_FLAG_WALL
-- Performing Test HPX_WITH_CXX_FLAG_WALL - Success
-- Performing Test HPX_WITH_C_FLAG_WALL
-- Performing Test HPX_WITH_C_FLAG_WALL - Success
-- Performing Test HPX_WITH_Fortran_FLAG_WALL
-- Performing Test HPX_WITH_Fortran_FLAG_WALL - Success
-- Performing Test HPX_WITH_CXX_FLAG_WEXTRA
-- Performing Test HPX_WITH_CXX_FLAG_WEXTRA - Success
-- Performing Test HPX_WITH_C_FLAG_WEXTRA
-- Performing Test HPX_WITH_C_FLAG_WEXTRA - Success
-- Performing Test HPX_WITH_Fortran_FLAG_WEXTRA
-- Performing Test HPX_WITH_Fortran_FLAG_WEXTRA - Success
-- Performing Test HPX_WITH_CXX_FLAG_WNO_UNUSED_LOCAL_TYPEDEFS
-- Performing Test HPX_WITH_CXX_FLAG_WNO_UNUSED_LOCAL_TYPEDEFS - Success
-- Performing Test HPX_WITH_C_FLAG_WNO_UNUSED_LOCAL_TYPEDEFS
-- Performing Test HPX_WITH_C_FLAG_WNO_UNUSED_LOCAL_TYPEDEFS - Success
-- Performing Test HPX_WITH_Fortran_FLAG_WNO_UNUSED_LOCAL_TYPEDEFS
-- Performing Test HPX_WITH_Fortran_FLAG_WNO_UNUSED_LOCAL_TYPEDEFS - Success
-- Performing Test HPX_WITH_CXX_FLAG_WNO_STRICT_ALIASING
-- Performing Test HPX_WITH_CXX_FLAG_WNO_STRICT_ALIASING - Success
-- Performing Test HPX_WITH_C_FLAG_WNO_STRICT_ALIASING
-- Performing Test HPX_WITH_C_FLAG_WNO_STRICT_ALIASING - Success
-- Performing Test HPX_WITH_Fortran_FLAG_WNO_STRICT_ALIASING
-- Performing Test HPX_WITH_Fortran_FLAG_WNO_STRICT_ALIASING - Success
-- Performing Test HPX_WITH_CXX_FLAG_WNO_SIGN_PROMO
-- Performing Test HPX_WITH_CXX_FLAG_WNO_SIGN_PROMO - Success
-- Performing Test HPX_WITH_CXX_FLAG_WNO_ATTRIBUTES
-- Performing Test HPX_WITH_CXX_FLAG_WNO_ATTRIBUTES - Success
-- Performing Test HPX_WITH_CXX_FLAG_WNO_CAST_ALIGN
-- Performing Test HPX_WITH_CXX_FLAG_WNO_CAST_ALIGN - Success
-- Performing Test HPX_WITH_CXX_FLAG_WNO_UNUSED_PARAMETER
-- Performing Test HPX_WITH_CXX_FLAG_WNO_UNUSED_PARAMETER - Success
-- Performing Test HPX_WITH_CXX_FLAG_WFORMAT_2
-- Performing Test HPX_WITH_CXX_FLAG_WFORMAT_2 - Success
-- Performing Test HPX_WITH_CXX_FLAG_WNO_FORMAT_NONLITERAL
-- Performing Test HPX_WITH_CXX_FLAG_WNO_FORMAT_NONLITERAL - Success
-- Performing Test HPX_WITH_CXX_FLAG_WINIT_SELF
-- Performing Test HPX_WITH_CXX_FLAG_WINIT_SELF - Success
-- Performing Test HPX_WITH_CXX_FLAG_WDOUBLE_PROMOTION
-- Performing Test HPX_WITH_CXX_FLAG_WDOUBLE_PROMOTION - Success
-- Performing Test HPX_WITH_CXX_FLAG_WCAST_QUAL
-- Performing Test HPX_WITH_CXX_FLAG_WCAST_QUAL - Success
-- Performing Test HPX_WITH_CXX_FLAG_WCAST_ALIGN
-- Performing Test HPX_WITH_CXX_FLAG_WCAST_ALIGN - Success
-- Performing Test HPX_WITH_CXX_FLAG_WERROR_TRAMPOLINES
-- Performing Test HPX_WITH_CXX_FLAG_WERROR_TRAMPOLINES - Success
-- Performing Test HPX_WITH_CXX_FLAG_WERROR_PARENTHESES
-- Performing Test HPX_WITH_CXX_FLAG_WERROR_PARENTHESES - Success
-- Performing Test HPX_WITH_CXX_FLAG_WERROR_REORDER
-- Performing Test HPX_WITH_CXX_FLAG_WERROR_REORDER - Success
-- Performing Test HPX_WITH_CXX_FLAG_WERROR_RETURN_TYPE
-- Performing Test HPX_WITH_CXX_FLAG_WERROR_RETURN_TYPE - Success
-- Performing Test HPX_WITH_CXX_FLAG_WERROR_SEQUENCE_POINT
-- Performing Test HPX_WITH_CXX_FLAG_WERROR_SEQUENCE_POINT - Success
-- Performing Test HPX_WITH_CXX_FLAG_WERROR_UNINITIALIZED
-- Performing Test HPX_WITH_CXX_FLAG_WERROR_UNINITIALIZED - Success
-- Performing Test HPX_WITH_CXX_FLAG_WERROR_FORMAT
-- Performing Test HPX_WITH_CXX_FLAG_WERROR_FORMAT - Success
-- Performing Test HPX_WITH_CXX_FLAG_WERROR_MISSING_BRACES
-- Performing Test HPX_WITH_CXX_FLAG_WERROR_MISSING_BRACES - Success
-- Performing Test HPX_WITH_CXX_FLAG_WERROR_SIGN_COMPARE
-- Performing Test HPX_WITH_CXX_FLAG_WERROR_SIGN_COMPARE - Success
-- Performing Test HPX_WITH_CXX_FLAG_FDIAGNOSTICS_SHOW_OPTION
-- Performing Test HPX_WITH_CXX_FLAG_FDIAGNOSTICS_SHOW_OPTION - Success
-- Performing Test HPX_WITH_C_FLAG_FDIAGNOSTICS_SHOW_OPTION
-- Performing Test HPX_WITH_C_FLAG_FDIAGNOSTICS_SHOW_OPTION - Success
-- Performing Test HPX_WITH_Fortran_FLAG_FDIAGNOSTICS_SHOW_OPTION
-- Performing Test HPX_WITH_Fortran_FLAG_FDIAGNOSTICS_SHOW_OPTION - Success
-- Performing Test HPX_WITH_CXX_FLAG_WERROR_VLA
-- Performing Test HPX_WITH_CXX_FLAG_WERROR_VLA - Success
-- Performing Test HPX_WITH_C_FLAG_WERROR_RETURN_TYPE
-- Performing Test HPX_WITH_C_FLAG_WERROR_RETURN_TYPE - Success
-- Performing Test HPX_WITH_CXX_FLAG_WNO_UNUSED_BUT_SET_PARAMETER
-- Performing Test HPX_WITH_CXX_FLAG_WNO_UNUSED_BUT_SET_PARAMETER - Success
-- Performing Test HPX_WITH_C_FLAG_WNO_UNUSED_BUT_SET_PARAMETER
-- Performing Test HPX_WITH_C_FLAG_WNO_UNUSED_BUT_SET_PARAMETER - Success
-- Performing Test HPX_WITH_CXX_FLAG_WNO_UNUSED_BUT_SET_VARIABLE
-- Performing Test HPX_WITH_CXX_FLAG_WNO_UNUSED_BUT_SET_VARIABLE - Success
-- Performing Test HPX_WITH_C_FLAG_WNO_UNUSED_BUT_SET_VARIABLE
-- Performing Test HPX_WITH_C_FLAG_WNO_UNUSED_BUT_SET_VARIABLE - Success
-- Performing Test HPX_WITH_C_FLAG_WERROR_UNINITIALIZED
-- Performing Test HPX_WITH_C_FLAG_WERROR_UNINITIALIZED - Success
-- Performing Test HPX_WITH_CXX_FLAG_WNO_MAYBE_UNINITIALIZED
-- Performing Test HPX_WITH_CXX_FLAG_WNO_MAYBE_UNINITIALIZED - Success
-- Performing Test HPX_WITH_C_FLAG_WNO_MAYBE_UNINITIALIZED
-- Performing Test HPX_WITH_C_FLAG_WNO_MAYBE_UNINITIALIZED - Success
-- Performing Test HPX_WITH_CXX_FLAG_WNO_SYNC_NAND
-- Performing Test HPX_WITH_CXX_FLAG_WNO_SYNC_NAND - Success
-- Performing Test HPX_WITH_C_FLAG_WNO_SYNC_NAND
-- Performing Test HPX_WITH_C_FLAG_WNO_SYNC_NAND - Success
-- Performing Test HPX_WITH_CXX_FLAG_WNO_DELETE_NON_VIRTUAL_DTOR
-- Performing Test HPX_WITH_CXX_FLAG_WNO_DELETE_NON_VIRTUAL_DTOR - Success
-- Performing Test WITH_LINKER_FLAG_WL__Z_DEFS
-- Performing Test WITH_LINKER_FLAG_WL__Z_DEFS - Success
-- Performing Test HPX_WITH_CXX_FLAG_FVISIBILITY_HIDDEN
-- Performing Test HPX_WITH_CXX_FLAG_FVISIBILITY_HIDDEN - Success
-- Performing Test HPX_WITH_C_FLAG_FVISIBILITY_HIDDEN
-- Performing Test HPX_WITH_C_FLAG_FVISIBILITY_HIDDEN - Success
-- Performing Test HPX_WITH_Fortran_FLAG_FVISIBILITY_HIDDEN
-- Performing Test HPX_WITH_Fortran_FLAG_FVISIBILITY_HIDDEN - Success
-- Performing Test WITH_LINKER_FLAG_FVISIBILITY_HIDDEN
-- Performing Test WITH_LINKER_FLAG_FVISIBILITY_HIDDEN - Success
-- Performing Test HPX_WITH_RDTSC - Success
-- Performing Test HPX_WITH_RDTSCP - Success
-- Boost version: 1.66.0
-- Found the following Boost libraries:
--   atomic
--   filesystem
--   program_options
--   regex
--   system
-- Found TCMalloc: /usr/lib/libtcmalloc_minimal.so  
-- Using tcmalloc allocator.
-- Found Hwloc: /usr/lib/libhwloc.so  
-- Could NOT find Git (missing: GIT_EXECUTABLE) 
-- WARNING: GIT commit not found (set to 'unknown').
-- Found PythonInterp: /usr/bin/python3.7 (found version "3.7.1") 
-- Found PkgConfig: /usr/bin/pkg-config (found version "1.5.3") 
-- Configuration summary:
--   HPX_WITH_AGAS_DUMP_REFCNT_ENTRIES=OFF
--   HPX_WITH_ALGORITHM_INPUT_ITERATOR_SUPPORT=OFF
--   HPX_WITH_APEX=OFF
--   HPX_WITH_ATTACH_DEBUGGER_ON_TEST_FAILURE=OFF
--   HPX_WITH_AUTOMATIC_SERIALIZATION_REGISTRATION=ON
--   HPX_WITH_BENCHMARK_SCRIPTS_PATH=/builddir/build/BUILD/hpx_1.2.0/serial/scripts
--   HPX_WITH_COMPILER_WARNINGS=ON
--   HPX_WITH_COMPILE_ONLY_TESTS=ON
--   HPX_WITH_COMPRESSION_BZIP2=OFF
--   HPX_WITH_COMPRESSION_SNAPPY=OFF
--   HPX_WITH_COMPRESSION_ZLIB=OFF
--   HPX_WITH_CUDA=OFF
--   HPX_WITH_CUDA_CLANG=OFF
--   HPX_WITH_CXX14_RETURN_TYPE_DEDUCTION=TRUE
--   HPX_WITH_DATAPAR_BOOST_SIMD=OFF
--   HPX_WITH_DATAPAR_VC=OFF
--   HPX_WITH_DEFAULT_TARGETS=ON
--   HPX_WITH_DEPRECATION_WARNINGS=ON
--   HPX_WITH_DISABLED_SIGNAL_EXCEPTION_HANDLERS=OFF
--   HPX_WITH_DOCUMENTATION=OFF
--   HPX_WITH_DYNAMIC_HPX_MAIN=ON
--   HPX_WITH_EXAMPLES=ON
--   HPX_WITH_EXAMPLES_HDF5=OFF
--   HPX_WITH_EXAMPLES_OPENMP=OFF
--   HPX_WITH_EXAMPLES_QT4=OFF
--   HPX_WITH_EXAMPLES_QTHREADS=OFF
--   HPX_WITH_EXAMPLES_TBB=OFF
--   HPX_WITH_FAIL_COMPILE_TESTS=ON
--   HPX_WITH_FAULT_TOLERANCE=OFF
--   HPX_WITH_FORTRAN=OFF
--   HPX_WITH_FULL_RPATH=ON
--   HPX_WITH_GCC_VERSION_CHECK=ON
--   HPX_WITH_GENERIC_CONTEXT_COROUTINES=OFF
--   HPX_WITH_GOOGLE_PERFTOOLS=OFF
--   HPX_WITH_HCC=OFF
--   HPX_WITH_HIDDEN_VISIBILITY=ON
--   HPX_WITH_HPXMP=OFF
--   HPX_WITH_INCLUSIVE_SCAN_COMPATIBILITY=ON
--   HPX_WITH_IO_COUNTERS=ON
--   HPX_WITH_IO_POOL=ON
--   HPX_WITH_ITTNOTIFY=OFF
--   HPX_WITH_LOGGING=ON
--   HPX_WITH_MALLOC=tcmalloc
--   HPX_WITH_MAX_CPU_COUNT=64
--   HPX_WITH_MAX_NUMA_DOMAIN_COUNT=4
--   HPX_WITH_MORE_THAN_64_THREADS=OFF
--   HPX_WITH_NATIVE_TLS=ON
--   HPX_WITH_NETWORKING=ON
--   HPX_WITH_NICE_THREADLEVEL=OFF
--   HPX_WITH_PAPI=OFF
--   HPX_WITH_PARCELPORT_ACTION_COUNTERS=OFF
--   HPX_WITH_PARCELPORT_LIBFABRIC=OFF
--   HPX_WITH_PARCELPORT_MPI=OFF
--   HPX_WITH_PARCELPORT_MPI_ENV=MV2_COMM_WORLD_RANK PMI_RANK OMPI_COMM_WORLD_SIZE ALPS_APP_PE
--   HPX_WITH_PARCELPORT_MPI_MULTITHREADED=ON
--   HPX_WITH_PARCELPORT_TCP=ON
--   HPX_WITH_PARCELPORT_VERBS=OFF
--   HPX_WITH_PARCEL_COALESCING=ON
--   HPX_WITH_PARCEL_PROFILING=OFF
--   HPX_WITH_PSEUDO_DEPENDENCIES=ON
--   HPX_WITH_QUEUE_COMPATIBILITY=OFF
--   HPX_WITH_RUNTIME=ON
--   HPX_WITH_RUN_MAIN_EVERYWHERE=OFF
--   HPX_WITH_SCHEDULER_LOCAL_STORAGE=OFF
--   HPX_WITH_SCOPED_UNLOCK_COMPATIBILITY=OFF
--   HPX_WITH_SPINLOCK_DEADLOCK_DETECTION=OFF
--   HPX_WITH_STACKOVERFLOW_DETECTION=OFF
--   HPX_WITH_STACKTRACES=ON
--   HPX_WITH_STATIC_LINKING=OFF
--   HPX_WITH_SWAP_CONTEXT_EMULATION=OFF
--   HPX_WITH_SYCL=OFF
--   HPX_WITH_TESTS=ON
--   HPX_WITH_TESTS_BENCHMARKS=ON
--   HPX_WITH_TESTS_EXAMPLES=ON
--   HPX_WITH_TESTS_EXTERNAL_BUILD=ON
--   HPX_WITH_TESTS_HEADERS=OFF
--   HPX_WITH_TESTS_REGRESSIONS=ON
--   HPX_WITH_TESTS_UNIT=ON
--   HPX_WITH_THREAD_BACKTRACE_DEPTH=5
--   HPX_WITH_THREAD_BACKTRACE_ON_SUSPENSION=OFF
--   HPX_WITH_THREAD_COMPATIBILITY=OFF
--   HPX_WITH_THREAD_CREATION_AND_CLEANUP_RATES=OFF
--   HPX_WITH_THREAD_CUMULATIVE_COUNTS=ON
--   HPX_WITH_THREAD_DEBUG_INFO=OFF
--   HPX_WITH_THREAD_DESCRIPTION_FULL=OFF
--   HPX_WITH_THREAD_GUARD_PAGE=ON
--   HPX_WITH_THREAD_IDLE_RATES=OFF
--   HPX_WITH_THREAD_LOCAL_STORAGE=OFF
--   HPX_WITH_THREAD_MANAGER_IDLE_BACKOFF=ON
--   HPX_WITH_THREAD_QUEUE_WAITTIME=OFF
--   HPX_WITH_THREAD_SCHEDULERS=all
--   HPX_WITH_THREAD_STACK_MMAP=ON
--   HPX_WITH_THREAD_STEALING_COUNTS=ON
--   HPX_WITH_THREAD_TARGET_ADDRESS=OFF
--   HPX_WITH_TIMER_POOL=ON
--   HPX_WITH_TOOLS=OFF
--   HPX_WITH_TUPLE_RVALUE_SWAP=ON
--   HPX_WITH_UNWRAPPED_COMPATIBILITY=ON
--   HPX_WITH_VALGRIND=OFF
--   HPX_WITH_VERIFY_LOCKS=OFF
--   HPX_WITH_VERIFY_LOCKS_BACKTRACE=OFF
--   HPX_WITH_VERIFY_LOCKS_GLOBALLY=OFF
--   HPX_WITH_VIM_YCM=OFF
--   HPX_WITH_ZERO_COPY_SERIALIZATION_THRESHOLD=128
HPX will be installed to /usr
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:
    CMAKE_C_FLAGS_RELEASE
    CMAKE_Fortran_FLAGS_RELEASE
    INCLUDE_INSTALL_DIR
    LIB_INSTALL_DIR
    SHARE_INSTALL_PREFIX
    SYSCONF_INSTALL_DIR
-- Build files have been written to: /builddir/build/BUILD/hpx_1.2.0/serial
```

```
CPU info:
Architecture:        i686
CPU op-mode(s):      32-bit, 64-bit
Byte Order:          Little Endian
CPU(s):              6
On-line CPU(s) list: 0-5
Thread(s) per core:  1
Core(s) per socket:  1
Socket(s):           6
NUMA node(s):        1
Vendor ID:           GenuineIntel
CPU family:          6
Model:               60
Model name:          Intel Core Processor (Haswell, no TSX)
Stepping:            1
CPU MHz:             2299.996
BogoMIPS:            4599.99
Hypervisor vendor:   KVM
Virtualization type: full
L1d cache:           32K
L1i cache:           32K
L2 cache:            4096K
L3 cache:            16384K
NUMA node0 CPU(s):   0-5
Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 syscall nx rdtscp lm constant_tsc rep_good nopl xtopology cpuid tsc_known_freq pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand hypervisor lahf_lm abm invpcid_single pti fsgsbase bmi1 avx2 smep bmi2 erms invpcid xsaveopt arat


Memory:
              total        used        free      shared  buff/cache   available
Mem:       15387432      786552     3552296         580    11048584    14260796
Swap:     134216700       56064   134160636


Storage:
Filesystem      Size  Used Avail Use% Mounted on
/dev/vda1       135G   14G  114G  11% /
```

--- a/examples/resource_partitioner/async_customization.cpp
+++ b/examples/resource_partitioner/async_customization.cpp
@@ -29,7 +29,7 @@
 #include "shared_priority_queue_scheduler.hpp"
 //
 #include <cstddef>
-#include <cstddef>
+#include <cstdint>
 #include <iostream>
 #include <memory>
 #include <string>
@@ -382,11 +382,11 @@ int test(Executor &exec)
     // test 3b
     std::cout << "============================" << std::endl;
     std::cout << "Test 3b : when_all(shared)" << std::endl;
-    future<uint64_t>     fws1 = make_ready_future(uint64_t(42));
+    future<std::uint64_t>     fws1 = make_ready_future(std::uint64_t(42));
     shared_future<float> fws2 = make_ready_future(3.1415f).share();
     //
     auto fws = when_all(fws1, fws2).then(exec,
-        [](future<util::tuple<future<uint64_t>, shared_future<float>>> && f)
+        [](future<util::tuple<future<std::uint64_t>, shared_future<float>>> && f)
         {
             auto tup = f.get();
             auto cmplx = std::complex<double>(
@@ -401,13 +401,13 @@ int test(Executor &exec)
     // test 4a
     std::cout << "============================" << std::endl;
     std::cout << "Test 4a : dataflow()" << std::endl;
-    future<uint16_t> f1 = make_ready_future(uint16_t(255));
+    future<std::uint16_t> f1 = make_ready_future(std::uint16_t(255));
     future<double>   f2 = make_ready_future(127.890);
     //
     auto fd = dataflow(exec,
-        [](future<uint16_t> && f1, future<double> && f2)
+        [](future<std::uint16_t> && f1, future<double> && f2)
         {
-            auto cmplx = std::complex<uint64_t>(f1.get(), f2.get());
+            auto cmplx = std::complex<std::uint64_t>(f1.get(), f2.get());
             std::cout << "Inside dataflow : " << cmplx << std::endl;
             return cmplx;
         }
@@ -419,13 +419,13 @@ int test(Executor &exec)
     // test 4b
     std::cout << "============================" << std::endl;
     std::cout << "Test 4b : dataflow(shared)" << std::endl;
-    future<uint16_t>      fs1 = make_ready_future(uint16_t(255));
+    future<std::uint16_t>      fs1 = make_ready_future(std::uint16_t(255));
     shared_future<double> fs2 = make_ready_future(127.890).share();
     //
     auto fds = dataflow(exec,
-        [](future<uint16_t> && f1, shared_future<double> && f2)
+        [](future<std::uint16_t> && f1, shared_future<double> && f2)
         {
-            auto cmplx = std::complex<uint64_t>(f1.get(), f2.get());
+            auto cmplx = std::complex<std::uint64_t>(f1.get(), f2.get());
             std::cout << "Inside dataflow(shared) : " << cmplx << std::endl;
             return cmplx;
         }
@@ -453,14 +453,14 @@ namespace hpx { namespace threads { namespace executors
           std::cout << "Hint 2 \n";
           return 0;
       }
-      int operator()(const uint16_t, const double) const {
+      int operator()(const std::uint16_t, const double) const {
           std::cout << "Hint 3 \n";
           return 0;
       }
       int operator()(const util::tuple<future<int>, future<double>> &) const {
           return 0;
       }
-      int operator()(const util::tuple<future<long unsigned int>,
+      int operator()(const util::tuple<future<std::uint64_t>,
                      shared_future<float>> &) const
       {
           return 0;

